server/ServerApp.cpp

```diff
@@ -1932,6 +1941,10 @@ int ServerApp::AddPlayerIntoGame(const PlayerConnectionPtr& player_connection, i
             const std::string previous_player_name = (*previous_it)->PlayerName();
             m_networking.Disconnect(previous_player_id);
             if (previous_client_type == Networking::ClientType::CLIENT_TYPE_AI_PLAYER) {
+                // change empire's player so after reload the player still could connect
+                // to the empire
```

Comment:
Does this let them reconnect to the same empire, or does it prevent other players from connecting to that empire? If you didn't set the name, couldn't the same player just reconnect to the same empire anyway...?

Reply:
Empire lost its AI client type so other player couldn't connect to it, but after reload empire will be started without assigned AI process so if it continue to have old `AI_#` player name no one could connect to it.